src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -1119,17 +1120,21 @@ public Solution WithProjectAnalyzerReferences(ProjectId projectId, IEnumerable<A
 
         private Solution AddDocument(DocumentState state)
         {
-            if (state == null)
+            var notificationService = this.Workspace.Services.GetService<IGlobalOperationNotificationService>();
+            using (notificationService?.Start("Remove Document"))
             {
-                throw new ArgumentNullException(nameof(state));
-            }
+                if (state == null)
+                {
+                    throw new ArgumentNullException(nameof(state));
+                }
 
-            CheckContainsProject(state.Id.ProjectId);
+                CheckContainsProject(state.Id.ProjectId);
```

Comment:
It looks like the project system calls AbsractProject.Disconnect() when a project is unloaded. We then start removing documents ourselves, so we can just start the global operation first.

When the project is reloaded, we see the file additions one at a time, so I'm not sure how to batch those.